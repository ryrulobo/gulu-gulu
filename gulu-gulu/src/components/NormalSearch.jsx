export default function NormalSearch({ results }) {
  return (
    <div className="flex flex-wrap justify-center space-y-5 sm:px-56 p-5">
      {results.map(({ displayUrl, name, snippet }, index) => (
        <div key={index} className="w-full">
          <a href={displayUrl} target="_blank" rel="noreferrer">
            {displayUrl && (
              <p className="text-sm">
                {displayUrl.length > 80
                  ? displayUrl.substring(0, 30)
                  : displayUrl}
              </p>
            )}
            <p className="text-lg hover:underline text-blue-700">{name}</p>
          </a>
          <p>{snippet}</p>
        </div>
      ))}
    </div>
  );
}
