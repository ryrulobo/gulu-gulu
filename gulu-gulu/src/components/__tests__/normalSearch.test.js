import { render, cleanup, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import NormalSearch from "../NormalSearch";

afterEach(() => {
  cleanup();
});

const results = [
  {
    displayUrl: "test.com/results",
    name: "Test title",
    snippet: "This is test title",
  },
];

test("Render normal search", () => {
  render(<NormalSearch results={results} />);
  const normalSearchElement = screen.getByTestId("normalSearch");
  expect(normalSearchElement).toBeInTheDocument();
  expect(normalSearchElement).toContainHTML("a");
  expect(normalSearchElement).toContainHTML("p");
  expect(normalSearchElement.children).toHaveLength(1);
});

test("Matches snapshots", () => {
  const tree = renderer.create(<NormalSearch results={results} />).toJSON();
  expect(tree).toMatchSnapshot();
});
