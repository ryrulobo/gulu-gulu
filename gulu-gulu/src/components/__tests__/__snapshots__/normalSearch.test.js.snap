// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshots 1`] = `
<div
  className="flex flex-wrap justify-center space-y-5 sm:px-56 p-5"
  data-testid="normalSearch"
>
  <div
    className="w-full"
  >
    <a
      href="test.com/results"
      rel="noreferrer"
      target="_blank"
    >
      <p
        className="text-sm"
      >
        test.com/results
      </p>
      <p
        className="text-lg hover:underline text-blue-700"
      >
        Test title
      </p>
    </a>
    <p>
      This is test title
    </p>
  </div>
</div>
`;
